bpfd/src/multiprog/tc.rs

```diff
@@ -110,19 +112,72 @@ impl TcDispatcher {
             loader: Some(loader),
         };
         dispatcher.attach_extensions(&mut extensions).await?;
-        dispatcher.attach(old_dispatcher)?;
+        dispatcher.attach(old_dispatcher).await?;
         dispatcher.save()?;
         Ok(dispatcher)
     }
 
-    fn attach(&mut self, old_dispatcher: Option<Dispatcher>) -> Result<(), BpfdError> {
+    /// has_qdisc returns true if the qdisc_name is found on the if_index.
+    async fn has_qdisc(qdisc_name: String, if_index: i32) -> Result<bool, anyhow::Error> {
+        let (connection, handle, _) = rtnetlink::new_connection().unwrap();
+        tokio::spawn(connection);
+
+        let mut qdiscs = handle.qdisc().get().execute();
```

Comment:
The QDiscHandle::get() method is equivalent to `tc qdisc show`.  So this works, but we get all the qdiscs on all the interfaces.  It would be nice if `rtnetlink` had a get option equivalent to `tc qdisc show dev <DEVICE>`.  It seems like it wouldn't be too hard to add, but this could be a future optimization.